pkg/webhook/policy.go

```diff
@@ -18,45 +17,26 @@ import (
 	authenticationv1 "k8s.io/api/authentication/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
-	k8sruntime "k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/serializer"
-	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
-	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission/builder"
 	atypes "sigs.k8s.io/controller-runtime/pkg/webhook/admission/types"
 )
 
 func init() {
-	AddToManagerFuncs = append(AddToManagerFuncs, AddPolicyWebhook)
+	CreateWebhookFuncs = append(CreateWebhookFuncs, AddValidatingWebhook)
 }
 
-const (
-	namespace = "gatekeeper-system"
-)
-
-var log = logf.Log.WithName("webhook")
-
-var (
-	runtimeScheme      = k8sruntime.NewScheme()
-	codecs             = serializer.NewCodecFactory(runtimeScheme)
-	deserializer       = codecs.UniversalDeserializer()
-	enableManualDeploy = flag.Bool("enable-manual-deploy", false, "allow users to manually create webhook related objects")
-	port               = flag.Int("port", 443, "port for the server. defaulted to 443 if unspecified ")
-	webhookName        = flag.String("webhook-name", "validation.gatekeeper.sh", "domain name of the webhook, with at least three segments separated by dots. defaulted to validation.gatekeeper.sh if unspecified ")
```

Comment:
Is there a reason why this flag was removed and is now a const? IMO the `validatingWHName` and `mutatingWHName` should be customizable by users.

Reply:
What is the use case for customizing the WH name?

Reply:
This value is returned as part of the admission response back to the users:

Error from server ([denied by pod-must-have-gk] you must provide labels: {"gatekeeper"}): error when creating "test.yaml": admission webhook "validation.gatekeeper.sh" denied the request: [denied by pod-must-have-gk] you must provide labels: {"gatekeeper"}

Reply:
I take your point.

Looking at kubebuilder 2, they no longer self-register webhooks.

https://book.kubebuilder.io/cronjob-tutorial/cert-manager.html

is it worth waiting until kubebuilder 2 is GA and customers can edit the deployment manifest directly rather than creating a config channel we know will be deprecated?

Reply:
IMO I think we should support this because Gatekeeper v2 had this capability. When we move to kubebuilder 2 then we can remove this since users customize this outside of Gatekeeper. WDYT?

Reply:
I'm not particularly enthusiastic about the flag proliferation, but if someone finds it useful, it's not that big of an issue for me.

Reply:
I removed the consts and added flags for the webhook names.